crypto/sm2/sm2_pmeth.c

```diff
@@ -135,16 +139,24 @@ static int pkey_sm2_encrypt(EVP_PKEY_CTX *ctx,
             return 1;
     }
 
-    return sm2_encrypt(ec, md, in, inlen, out, outlen);
+    fetched_md = EVP_MD_fetch(libctx, EVP_MD_name(md), 0);
```

Comment:
Er...  if `md = dctx->md`, isn't it already fetched?

Reply:
Not sure: it does this.. How does this tell you its fetched?    
  case EVP_PKEY_CTRL_MD:
        smctx->md = p2;
        return 1;